src/metabase/lib/js.cljs

```diff
@@ -248,8 +248,9 @@
   looking at the last stage.
 
   > **Code health:** Healthy"
-  [a-query stage-number]
-  (if (and
+  [a-query stage-number card-id]
+  (let [a-query (lib.drill-thru.common/prepare-query a-query stage-number card-id)]
```

Comment:
`prepare-query` should go somewhere more common, with a better name. It needs to return the updated stage number as well, I suppose.

Reply:
Probably a direct function that returns `{:query ..., :stage-number ...}`, but also most of the call sites want to call it in `update` style. So perhaps something like

`(lib.common/with-mbql-stage query stage-number card-id lib/some-other-fn arg1 arg2)` which calls `prepare-query` with the first three args, then uses the updated `query` and `stage-number` it returns to do `(lib/some-other-fn query' stage-number' arg1 arg2)`.